collection-scripts/gather_network_logs

```diff
@@ -228,21 +228,41 @@ function gather_kuryr_data {
 
 function gather_kuryr_nodes_data {
   echo "INFO: Gathering kuryr nodes data"
-  for NODE in ${NODES}; do
-      CNI_POD=$(oc -n openshift-kuryr get pods --no-headers -o custom-columns=":metadata.name" --field-selector spec.nodeName="${NODE}" -l app=kuryr-cni)
+  for NODE in "${NODES[@]}"; do
+      CNI_POD=($(oc -n openshift-kuryr get pods --no-headers -o custom-columns=":metadata.name" --field-selector spec.nodeName="${NODE}" -l app=kuryr-cni))
       oc exec -n openshift-kuryr "${CNI_POD}" -- bash -c "for pid in $(find /host_proc/[1-9]*/ns/net | cut -d/ -f3); do \
           echo 'Namespace: \$pid'; \
           nsenter --net=/host_proc/\$pid/ns/net ip -d addr; \
           done" > "${NETWORK_LOG_PATH}"/"${CNI_POD}"_interfaces & PIDS+=($!)
   done
 }
-
-if [ $# -eq 0 ]; then
+ 
+ARGS=($@) #args passed in by the user
+IFS=" " read -r -a NODES <<< $(oc get nodes -o jsonpath='{range .items[*]}{@.metadata.name} {.status.nodeInfo.operatingSystem==linux}')
+ELEMENTS=${#ARGS[@]} #length of args passed in by the user
+NODE_LENGTH=${#NODES[@]}
+TMP_ARR=() #tmp array to place the args that are nodes when they are checked against the cluster nodes
+if [ $# -eq 0 ]; then 
     echo "WARNING: Collecting network logs on ALL linux nodes in your cluster. This could take a long time." >&2
+else
+    for (( i=0; i<ELEMENTS; i++));do
+      for (( j=0; j<NODE_LENGTH; j++));do
+        if [[ "${ARGS[$i]}" == "${NODES[$j]}" ]]; then
+        echo  ${ARGS[$i]} "->node exists in nodes"
+        TMP_ARR[${#TMP_ARR[@]}]+="${NODES[$j]}"
```

Comment:
Why do you need the TMP_ARR array here? Just remove it from the `ARGS` array if it doesn't exist in `NODES
`